google/cloud/bigquery/dataset.py

```diff
@@ -542,7 +555,20 @@ def to_api_repr(self):
         Returns:
             Dict[str, object]: Access entry represented as an API resource
         """
-        resource = copy.deepcopy(self._properties)
+        resource = {}
+
+        if self.role is not None:
```

Comment:
Why can't we continue using `copy.deepcopy()` here? I suppose it should take care of any object that needs to be duplicated recursively.